From 9180b53c80fbe7f8c43850ca9ccbdd3e42368866 Mon Sep 17 00:00:00 2001
From: Alex Wilson <alex.wilson@joyent.com>
Date: Mon, 22 Aug 2016 18:19:36 -0700
Subject: [PATCH] AGENT-1029 CN agent reports negative disk_system_used_bytes
 values in sysinfo Reviewed by: Joshua M. Clulow <jmc@joyent.com> Reviewed by:
 Orlando Vazquez <orlando@joyent.com> Approved by: Orlando Vazquez
 <orlando@joyent.com>

---
 lib/heartbeater.js | 18 +++++-------------
 1 file changed, 5 insertions(+), 13 deletions(-)

diff --git a/lib/heartbeater.js b/lib/heartbeater.js
index 24f303a..c8b7a92 100644
--- a/lib/heartbeater.js
+++ b/lib/heartbeater.js
@@ -518,19 +518,11 @@ StatusReporter.prototype.gatherDiskUsage = function (vms, callback) {
         },
         function (cb) {
             // #7
-            zpool.list(
-                'zones', { parseable: true, fields: ['size', 'alloc'] },
-                function (error, fields, pools) {
-                    if (error) {
-                        cb(error);
-                        return;
-                    }
-                    usage.pool_size_bytes = toInt(pools[0][0]);
-                    usage.pool_alloc_bytes = toInt(pools[0][1]);
-                    cb();
-                });
-        },
-        function (cb) {
+            var poolds = datasets['zones'];
+            usage.pool_alloc_bytes = toInt(poolds.used);
+            usage.pool_size_bytes = toInt(poolds.used) +
+                toInt(poolds.available);
+
             // #8
             // All separated usages should be subtracted from the allocated
             // here. Anything not specifically listed is treated as 'system
-- 
2.21.0

